### DISCUSSION TITLE
[Cache] Multiple Redis drivers hit the same key and it takes time

### DISCUSSION THREAD

**Discussion by egyjs** [laravel/framework#38309] [Ideas]
![Cache . Multiple Redis drivers hit the same key and it takes time](https://user-images.githubusercontent.com/12745270/128801772-8aed2a78-f17a-40ae-8247-3271ae7860cf.png)

As you can see in the image here, we are calling the same key multiple times in the same request.
In each case, it takes `0.16` or even `0.26` milliseconds. As a result, we will see that even if the caches on the same key are many, it will take quite some time
....

so i was trying something, if i could I  override the method `get()` of the `class Illuminate\Cache\Repository` as in this: [Q](https://stackoverflow.com/q/68686644/7511165) 

I will be able to put any keys that already have been hited  in a php array in in custom class `CacheHandler`  for example : 
```php
<?php

namespace App\Lib\CacheHandler;

class CacheHandler
{
    public $stats = [];

    /**
     * @param $key
     * @param $value
     */
    public function setStats($key, $value): void
    {
        $this->stats[$key] = $value;
    }

    /**
     * @param $key
     * @return mixed|null
     */
    public function getStats($key)
    {
        //return $this->stats[$key];
        //return $this->stats[$key] ?? null; // can't make `nothingHere` as null  because  stats[$key] = null
        return $this->stats[$key] ?? 'nothingHere'; // can't make `nothingHere` as null  because  stats[$key] = null
    }


}
```

**Comment by dennisprudlo** [comment]
Sometimes we want to hit the cache multiple times. Say I want to read a key, update it and another part of the app (but same request) needs to retrieve the updated entry. Caching the cache in php memory would always return the old value except you override all other methods as well (`set`, `pull`, ...)

`pull` for instance uses `get` but also forgets the key after reading. So when overriding `get` to read once and store it in array we would get wrong results.


```php
$store = [];

cache()->get('foo'); // hits the cache and stores result in $store
// $store is now: [ 'foo' => 'bar' ]
```
When using `cache()->get('foo')` again php uses the value from `$store` variable instead of hitting the cache. But when using `pull` the behavior is wrong.

```php
// cache has: 'baz' => 'hello'

cache()->pull('baz'); // hits cache, stores in $store variable, removes key from cache
// hello

cache()->get('baz'); // checks $store variable, returns value from variable
// hello
```

The `get` on the cache returns the value `hello` but it should be `null` (the cache key `baz` was removed from the cache).
